Homeworks/Homework W8/SQLQuery_HomeworkW8.sql

```diff
@@ -0,0 +1,340 @@
+INSERT INTO ZoneArea
+VALUES ('Fish');
+INSERT INTO ZoneArea
+VALUES ('Amphibians');
+INSERT INTO ZoneArea
+VALUES ('Reptiles');
+INSERT INTO ZoneArea
+VALUES ('Birds');
+INSERT INTO ZoneArea
+VALUES ('Mammals');
+
+SELECT *
+FROM ZoneArea;
+
+INSERT INTO Species
+VALUES ('Goldfish',1);
+INSERT INTO Species
+VALUES ('Angelfish',1);
+INSERT INTO Species
+VALUES ('Neon Tetra',1);
+INSERT INTO Species
+VALUES ('Catfish',1);
+INSERT INTO Species
+VALUES ('Frog',2);
+INSERT INTO Species
+VALUES ('Toad',2);
+INSERT INTO Species
+VALUES ('Salamander',2);
+INSERT INTO Species
+VALUES ('Newt',2);
+INSERT INTO Species
+VALUES ('Turtle',3);
+INSERT INTO Species
+VALUES ('Lizard',3);
+INSERT INTO Species
+VALUES ('Snake',3);
+INSERT INTO Species
+VALUES ('Peafowl',4);
+INSERT INTO Species
+VALUES ('Emu',4);
+INSERT INTO Species
+VALUES ('Parrot',4);
+INSERT INTO Species
+VALUES ('Duck',4);
+INSERT INTO Species
+VALUES ('Swan',4);
+INSERT INTO Species
+VALUES ('Tiger',5);
+INSERT INTO Species
+VALUES ('Gorilla',5);
+INSERT INTO Species
+VALUES ('Leopard',5);
+INSERT INTO Species
+VALUES ('Bear',5);
+
+
+
+SELECT *
+FROM Species;
+
+SELECT *
+FROM ZoneArea;
+
+
+
+ALTER TABLE Workers
+DROP constraint FK__Workers__WorkAre__3F466844;
+ALTER TABLE Workers
+DROP COLUMN WorkArea;
+
+ALTER TABLE ZoneArea
+ADD Worker INT REFERENCES Workers(Id);
+
+SELECT *
+FROM Workers;
+
+INSERT INTO Workers
+VALUES ('Gigi','Negoi');
+INSERT INTO Workers
+VALUES ('Liviu','Albu');
+INSERT INTO Workers
+VALUES ('Lavinia','Sagan');
+INSERT INTO Workers
+VALUES ('Luca','Nifor');
+
+-- modifing table ZoneArea after Changing foreign key
+UPDATE ZoneArea
+SET Worker = 4 
+WHERE Id = 5;
+
+SELECT *
+FROM ZoneArea;
+-- Safety code to run tables and see if joins are working
+SELECT z.[Zone], w.FirstName
+FROM ZoneArea z
+JOIN Workers  w ON z.Worker=w.Id;
+
+SELECT * 
+FROM Ticket;
+
+INSERT INTO Ticket
+VALUES(10,'2023-01-20','2023-01-20',1);
+INSERT INTO Ticket
+VALUES(10,'2023-01-20','2023-01-20',2);
+INSERT INTO Ticket
+VALUES(50,'2023-01-20','2023-01-27',3);
+INSERT INTO Ticket
+VALUES(80,'2023-01-20','2023-02-20',2);
+INSERT INTO Ticket
+VALUES(50,'2023-02-10','2023-02-20',2,4);
+INSERT INTO Ticket
+VALUES(10,'2023-01-22','2023-01-22',4,5);
+INSERT INTO Ticket
+VALUES(80,'2023-01-24','2023-02-24',5,6);
+
+ALTER TABLE Visitors
+DROP constraint FK__Visitors__Ticket__44FF419A;
+ALTER TABLE Visitors
+DROP COLUMN Ticket;
+
+ALTER TABLE Ticket
+ADD Visitor INT REFERENCES Visitors(Id);
+
+SELECT * 
+FROM Ticket;
+
+UPDATE Ticket
+SET Visitor = 3 
+WHERE Id = 9;
+
+SELECT *
+FROM Visitors;
+SELECT * 
+FROM Ticket;
+
+INSERT INTO Visitors
+VALUES('Alexandru','Mihai');
+INSERT INTO Visitors
+VALUES('Loredana','Luca');
+INSERT INTO Visitors
+VALUES('Elvis','Musteata');
+INSERT INTO Visitors
+VALUES('Emanuela','Coroi');
+INSERT INTO Visitors
+VALUES('George','Buhusi');
+INSERT INTO Visitors
+VALUES('Gina','Doroh');
+
+INSERT INTO Animals
+VALUES (1);
```

Comment:
According to the Animals table structure each animal should also have a SpeciesId. Also a name would be nice to have because in real life animals are known as "Otto the lion" and not only by "1 the lion". :)